src/server/gc_worker/gc_worker.rs

```diff
@@ -422,6 +454,139 @@ where
         Ok((handled_keys, wasted_keys))
     }
 
+    fn raw_gc_keys(
+        &mut self,
+        keys: Vec<Key>,
+        safe_point: TimeStamp,
+        regions_provider: Option<(u64, Arc<dyn RegionInfoProvider>)>,
+    ) -> Result<(usize, usize)> {
+        let first_key = keys.first().unwrap();
+        let first_raw_key = first_key.clone().append_ts(TimeStamp::max());
```

Comment:
The naming of the variables doesn't make sense to me, and it seems the only usage of `first_raw_key` and `last_raw_key` is to create a slice reference. It's enough to keep the cloned key in owned form only.